packages/contracts-bedrock/contracts/test/L1Block.t.sol

```diff
@@ -91,5 +105,141 @@ contract L1BlockTest is CommonTest {
+        // setup so we are expecting a revert here
+    }
+
+    function test_getL1BlockHash_lowerBound_l1BlockHashNumberOutOfBounds_reverts() external {
+        depositor = lb.DEPOSITOR_ACCOUNT();
+        vm.prank(depositor);
+        lb.setL1BlockValues({
+            _number: uint64(BLOCK_HASHES_SIZE + 1),
+            _timestamp: uint64(2),
+            _basefee: 3,
+            _hash: NON_ZERO_HASH,
+            _sequenceNumber: uint64(4),
+            _batcherHash: bytes32(0),
+            _l1FeeOverhead: 2,
+            _l1FeeScalar: 3
+        });
+
+        vm.expectRevert(L1BlockHashNumberOutOfBounds.selector);
+        lb.getL1BlockHash(1);
+    }
+
+    function test_getL1BlockHash_hashNotSet_l1BlockHashNumberOutOfBounds_reverts() external {
+        depositor = lb.DEPOSITOR_ACCOUNT();
+        vm.prank(depositor);
+        lb.setL1BlockValues({
+            _number: uint64(3),
+            _timestamp: uint64(2),
+            _basefee: 3,
+            _hash: bytes32(0),
+            _sequenceNumber: uint64(4),
+            _batcherHash: bytes32(0),
+            _l1FeeOverhead: 2,
+            _l1FeeScalar: 3
+        });
+
+        vm.expectRevert(L1BlockHashNumberOutOfBounds.selector);
+        lb.getL1BlockHash(2);
+    }
+
+    function test_getL1BlockHash_previousBlock_succeeds() external {
+        depositor = lb.DEPOSITOR_ACCOUNT();
+        vm.startPrank(depositor);
+        lb.setL1BlockValues({
+            _number: uint64(2),
+            _timestamp: uint64(2),
+            _basefee: 3,
+            _hash: keccak256(abi.encode(2)),
+            _sequenceNumber: uint64(4),
+            _batcherHash: bytes32(0),
+            _l1FeeOverhead: 2,
+            _l1FeeScalar: 3
+        });
+
+        lb.setL1BlockValues({
+            _number: uint64(3),
+            _timestamp: uint64(2),
+            _basefee: 3,
+            _hash: keccak256(abi.encode(3)),
+            _sequenceNumber: uint64(4),
+            _batcherHash: bytes32(0),
+            _l1FeeOverhead: 2,
+            _l1FeeScalar: 3
+        });
+        vm.stopPrank();
+
+        assertEq(lb.getL1BlockHash(2), keccak256(abi.encode(2)));
+        assertEq(lb.getL1BlockHash(3), keccak256(abi.encode(3)));
+    }
+
+    function test_setL1BlockValues_nonDepositor_reverts() external {
+        address nonDepositor = vm.addr(1234);
+        vm.prank(nonDepositor);
+        vm.expectRevert("L1Block: only the depositor account can set L1 block values");
+        lb.setL1BlockValues({
+            _number: type(uint64).max,
+            _timestamp: type(uint64).max,
+            _basefee: type(uint256).max,
+            _hash: keccak256(abi.encode(1)),
+            _sequenceNumber: type(uint64).max,
+            _batcherHash: bytes32(type(uint256).max),
+            _l1FeeOverhead: type(uint256).max,
+            _l1FeeScalar: type(uint256).max
+        });
+    }
+
+    function test_getL1BlockHash_succeeds() external {
```

Comment:
we could change this to a fuzz test where we have a BN provided as an argument and assume that the value is > n* BLOCK HASHES SIZE, where n is the number of times we want to "loop" the ring buffer.
We can add the `setL1BLockValues` in a for loop where we populate the block values starting from `value - BLOCK_HASHES_SIZE` until `value`. 

For each populated value we can check that the correct l1 block values have been populated.

Reply:
having this as a fuzz test takes too much time